Update flux_led for addition of config flow (#19566)
home-assistant/home-assistant.github.io: source/_integrations/flux_led.markdown

  - light
---

> The Flux Led integration supports several brands of both bulbs and controllers that use the same protocol and have the HF-LPB100 chipset in common. Chances are high that your bulb or controller (eg. WiFi LED CONTROLLER) will work with this integration if you can control the device with the MagicHome app.

Reviewer: ```suggestion
The Flux LED integration supports several brands of both bulbs and controllers that use the same protocol and have the HF-LPB100 chipset in common. Chances are high that your bulb or controller (eg. WiFi LED CONTROLLER) will work with this integration if you can control the device with the MagicHome app.
```

> The `Flux_LED` support is integrated into Home Assistant as a light platform. Several brands of both bulbs and controllers use the same protocol and they have the HF-LPB100 chipset in common. The chances are high that your bulb or controller (eg. WiFi LED CONTROLLER) will work if you can control the device with the MagicHome app.

Reviewer: We can remove the `Flux_LED` and just use normal words now, as this is no longer a platform. Also, remove the platform reference in the first sentence for the same reason.

- [Ledenet WiFi RGBW Controller](https://amzn.to/2WZKXNa)
- [SUPERNIGHT WiFi Wireless LED Smart Controller](https://amzn.to/2WURx7w)

{% include integrations/config_flow.md %}

> After the devices have been added they can be configured with different effects listed below. These setting can be accessed by navigating to the integration settings in configuration->integrations and selecting the "Flux Led/Magic Home" configuration for the bulb or controller. 

Reviewer: ```suggestion
After the devices have been added they can be configured with different effects listed below. These settings can be accessed by navigating to the integration settings in Configuration -> Integrations and selecting the "Flux Led/Magic Home" configuration for the bulb or controller. 
```

- [Ledenet WiFi RGBW Controller](https://amzn.to/2WZKXNa)
- [SUPERNIGHT WiFi Wireless LED Smart Controller](https://amzn.to/2WURx7w)

{% include integrations/config_flow.md %}

> After the devices have been added they can be configured with different effects or modes listed below. By default `Flux_LED` will attempt to determine the device type automatically. If it does not detect the correct device type it is recommended that the mode be selected manually.

Reviewer: ```suggestion
After the devices have been added they can be configured with different effects or modes listed below. By default, the integration will attempt to determine the device type automatically. If it does not detect the correct device type it is recommended that the mode be selected manually.
```

**Brightness Mode:**\

> **Auto** - `Flux_LED` will attempt to determine the mode of the device.\

Reviewer: This is a bit unclear to me as a reader, where do I set this?

Author: The modes are no longer relevant so I removed them and hopefully cleared up where to change the settings at.

**RGB** - Device supports three LED channels (**R**)ed, (**G**)reen, (**B**)lue\
**RGBW** - Device supports four LED channels (**R**)ed, (**G**)reen, (**B**)lue, (**W**)hite\
**W** - Device supports one LED channel (**W**)hite

**Custom Effect**\
A list of RGB colors can be entered to create an effect. The effect speed can be adjusted using the slider underneath.

**Custom Effect Type**\

> This determins the transition between each color. 

Reviewer: ```suggestion
This determines the transition between each color. 
```

- [Ledenet WiFi RGBW Controller](https://amzn.to/2WZKXNa)
- [SUPERNIGHT WiFi Wireless LED Smart Controller](https://amzn.to/2WURx7w)

> ### Configuration

Reviewer 2: ```suggestion
```

The include will provide the header

> The `Flux_LED` integration can be configured in the UI.

Reviewer 2: ```suggestion
```

The include will provide this

| `red_strobe`, `green_strobe`, `blue_strobe`, `yellow_strobe`, `cyan_strobe`, `purple_strobe`, `white_strobe` | Strobes the color indicated by the effect name.                    |
| `random`                                                                                                     | Chooses a random color by selecting random values for R, G, and B. |

### Custom Effects - Service `flux_led.set_custom_effect`

The `Flux_LED` integration offers a custom service to enable you to set the lights to a custom light effect. 

| Service data attribute | Description |
| ---------------------- | ----------- |
| `entity_id` | The entity_id of the LED light to set the effect on. |
| `colors` | List of RGB colors to transition between in your effect. (Max 16, Required) |
| `speed_pct` | The speed of the effect in % (0-100. Default 50) |
| `transition` | The transition effect you would like. Valid options are `gradual`, `jump`, or `strobe`. (Default `gradual`) |

```yaml
#Example Service Call
entity_id: light.led_strip
colors:
  - [255,0,0]
  - [0,255,0]
  - [0,0,255]
speed_pct: 80
transition: jump
```

> ### Depreceated configuration.yaml

Reviewer 2: Usually we remove the entire yaml configuration information once its deprecated to prevent users from going down a path that will shortly not exist.

Author: I wasn't sure.  I'll remove it.  I also added % include integrations/config_flow.md %

- [Ledenet WiFi RGBW Controller](https://amzn.to/2WZKXNa)
- [SUPERNIGHT WiFi Wireless LED Smart Controller](https://amzn.to/2WURx7w)

> ### Configuration
> 
> The `Flux_LED` integration can be configured in the UI or in the configuration.yaml.  Please see the configuration.yaml section for details on using the 'configuration.yaml' to configure `Flux_LED`. `
> 
> % include integrations/config_flow.md %
> 

Reviewer 2: ```suggestion
{% include integrations/config_flow.md %}
```

| `gb_cross_fade`                                                                                              | Fades between green and blue.                                      |
| `red_strobe`, `green_strobe`, `blue_strobe`, `yellow_strobe`, `cyan_strobe`, `purple_strobe`, `white_strobe` | Strobes the color indicated by the effect name.                    |
| `random`                                                                                                     | Chooses a random color by selecting random values for R, G, and B. |

### Custom Effects - Service `flux_led.set_custom_effect`

> The `Flux_LED` integration offers a custom service to enable you to set the lights to a custom light effect. 

Reviewer: ```suggestion
The integration offers a custom service to enable you to set the lights to a custom light effect. 
```

| Service data attribute | Description |
| ---------------------- | ----------- |
| `entity_id` | The entity_id of the LED light to set the effect on. |
| `colors` | List of RGB colors to transition between in your effect. (Max 16, Required) |
| `speed_pct` | The speed of the effect in % (0-100. Default 50) |
| `transition` | The transition effect you would like. Valid options are `gradual`, `jump`, or `strobe`. (Default `gradual`) |

```yaml
#Example Service Call
entity_id: light.led_strip
colors:
  - [255,0,0]
  - [0,255,0]
  - [0,0,255]
speed_pct: 80

> transition: jump

Reviewer: ```suggestion
transition: "jump"
```
